contentcuration/contentcuration/frontend/channelEdit/vuex/clipboard/actions.js

```diff
@@ -89,76 +90,150 @@ export function loadClipboardNodes(context, { parent, ancestorId }) {
+  }
+
+  if (contentNode && contentNode.resource_count > 0) {
+    return context
+      .dispatch('contentNode/loadContentNodes', { parent: contentNode.id }, { root: true })
+      .then(nodes => {
+        // Ensure we don't add excluded nodes to our Vuex map
+        nodes = nodes
+          .filter(node => !isExcludedNode(ancestor, node))
+          .map(node => ({
+            // For new version clipboard nodes, we don't need `node.id` for anything
+            // so we can change it, and the clipboard should not be manipulating the
+            // source node anyway, normally requiring its ID
+            id: selectionId(node.id, ancestor.id),
+            kind: node.kind,
+            parent,
+            lft: node.lft,
+            source_node_id: node.node_id,
+            source_channel_id: node.channel_id,
+            total_count: node.total_count,
+            resource_count: node.resource_count,
+            extra_fields: {
+              [ClipboardNodeFlag]: true,
+            },
+          }));
+
+        return context.dispatch('addClipboardNodes', {
+          nodes,
+          parent,
         });
-    }
+      });
   }
   return [];
 }
 
+export function addClipboardNodes(context, { nodes, parent }) {
+  const isRootInsert = parent === context.rootGetters['clipboardRootId'];
+  nodes.forEach(node => {
+    // Add selection state
+    if (!(node.id in context.state.selected)) {
+      // This is referencing what the "parent" is in the clipboard UI for determining
+      // selection inheritance
+      const selectionParent = isRootInsert ? node.source_channel_id : node.parent;
+
+      // Determine parent selection state so we can add new descendant nodes
+      // with the appropriate selection state
+      const parentSelection = context.getters.currentSelectionState(selectionParent);
+
+      context.commit('UPDATE_SELECTION_STATE', {
+        id: node.id,
+        selectionState:
+          parentSelection & SelectionFlags.ALL_DESCENDANTS ? SELECT_ALL : SelectionFlags.NONE,
+      });
+    }
+  });
+
+  context.commit('ADD_CLIPBOARD_NODES', nodes);
+  return nodes;
+}
+
+/**
+ * Dedicated Vuex action for loading nodes from the IndexedDB listener
+ */
+export function addClipboardNodeFromListener(context, obj) {
+  let promise = Promise.resolve();
+
+  // Load channel and create color if not already present on the clipboard
+  if (!context.getters.channelIds.includes(obj.source_channel_id)) {
+    promise = context.dispatch('loadChannels', { id__in: [obj.source_channel_id] });
+  }
+
+  return promise
+    .then(() =>
+      context.dispatch(
+        'contentNode/loadContentNodes',
+        { '[node_id+channel_id]__in': [obj.source_node_id, obj.source_channel_id] },
+        { root }
+      )
+    )
+    .then(() =>
+      context.dispatch('addClipboardNodes', {
+        nodes: [obj],
+        parent: obj.parent,
+      })
+    )
+    .then(() => context.dispatch('loadChannelColors'));
+}
```

Comment:
Ensures that channels and content nodes required for clipboard display are loaded into their respective Vuex states, and that we trigger creating a new channel color for new channels added to the clipboard while it's open. This gets triggered from IndexedDB create events on the clipboard table.